images/win/scripts/Installers/Install-SQLExpress.ps1

```diff
@@ -0,0 +1,114 @@
+    }
+    else
+    {
+      Write-Host -Object "Full package downloading process was unsuccessful. Exit code: $exitCode."
+      exit $exitCode
+    }
+}
+
+function Unpack-SQLInstaller {
+    param(
+        [String]$InstallPath,
+        [String]$Arguments = ("/Q", "/IACCEPTSQLSERVERLICENSETERMS")
+    )
+    $process = Start-Process -FilePath $InstallPath -ArgumentList $Arguments -Wait -PassThru
+    $exitCode = $process.ExitCode
+# Exit code -2067529716 is added since SQL Unpack procedure returns it on success.
+    if ($exitCode -eq 0 -or $exitCode -eq -2067529716)
+    {
+      Write-Host -Object "SQL installer unpacking has been completed."
+    }
+    else
+    {
+      Write-Host -Object "SQL installer unpacking was interrupted : $exitCode."
+      exit $exitCode
+    }
+}
+
+function Start-Installer {
+    param(
+        [String]$InstallPath,
+        [String]$Arguments = ("/Q", "/IACCEPTSQLSERVERLICENSETERMS", "/Action=Install", "/INSTANCEID=SQL2019", "/INSTANCENAME=SQL2019", "/SECURITYMODE=SQL", "/SAPWD=P@ssword!!", "/TCPENABLED=1")
+    )
+    $process = Start-Process -FilePath $InstallPath -ArgumentList $Arguments -Wait -PassThru
+    $exitCode = $process.ExitCode
+# Exit code -2067529716 is added since SQL Unpack procedure returns it on success.
+    if ($exitCode -eq 0)
+    {
+      Write-Host -Object "SQL Express has been successfully installed: ExitCode: $exitCode"
+    }
+    else
+    {
+      Write-Host -Object "Installation procedure was not correctly completed. Exit code: $exitCode."
+      exit $exitCode
+    }
+}
+
+
+function Start-DownloadSQLExpress {
+    param(
+        [String]$InstallerUrl,
+        [String]$InstallerPath,
+        [int] $retries = 10
+    )
+    Write-Host "Downloading SQL Express web-installer from: $InstallerUrl"
+    # In case of any errors -try to download package several times
+    while($retries -gt 0)
+        {
+            try
+            {
+                Write-Host "Downloading SQL Express installer..."
+                (New-Object System.Net.WebClient).DownloadFile($InstallerUrl, $InstallerPath)
+                break
+            }
+            catch
+            {
+                Write-Host "There is an error during SQL Express downloading"
+                $_
+
+                $retries--
+
+                if ($retries -eq 0)
+                {
+                    Write-Host "File can't be downloaded"
+                    $_
+                    exit 1
+                }
+
+                Write-Host "Waiting 30 seconds before retrying. Retries left: $retries"
+                Start-Sleep -Seconds 30
+            }
+        }
+    Write-Host "SQL Express has been successfully downladed from the link: $InstallerUrl"
+}
+#default variables for functions.
+$installerUrl = "https://go.microsoft.com/fwlink/?linkid=866658"
+$installerPath = "${env:Temp}\SQL2019-SSEI-Expr.exe"
+$downloadPath = "C:\SQLEXPRESS2019"
```

Comment:
Just to clarify installation process:
1) We download SQL Express package to $env:TEMP
2) Download full SQL Package to "C:\SQLEXPRESS2019"
3) Unpack full SQL Package in "C:\SQLEXPRESS2019"
4) Install it

Do we need to clean up "C:\SQLEXPRESS2019" and remove installers?
Is it possible to use Temp folder for all installation files? In this case, we won't need to take care about clean up